### ISSUE TITLE
FileNotFoundException when running test with quoteOutputMatches

### ISSUE THREAD

**Issue by ocampoleandro** [arrow-kt/arrow-meta#760]
**Describe the bug**
When using quoteOutputMatches in a unit test, the test fails with FileNotFoundException.
In my case

```
java.io.FileNotFoundException: /var/folders/2c/v1qydxqd3h5fdlqzb671vkhr9lc6zs/T/Kotlin-Compilation3027256973569121546/sources/Source.kt.meta (No such file or directory)

	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:125)
	at kotlin.io.FilesKt__FileReadWriteKt.readText$default(FileReadWrite.kt:125)
	at arrow.meta.plugin.testing.CompilationAssertionsKt.assertQuoteFileMatches(CompilationAssertions.kt:173)
	at arrow.meta.plugin.testing.CompilationAssertionsKt.assertQuoteOutputMatches(CompilationAssertions.kt:152)
	at arrow.meta.plugin.testing.CompilationAssertionsKt.access$assertQuoteOutputMatches(CompilationAssertions.kt:1)
	at arrow.meta.plugin.testing.CompilationAssertionsKt$interpreter$1$4.invoke(CompilationAssertions.kt:98)
	at arrow.meta.plugin.testing.CompilationAssertionsKt$interpreter$1.invoke(CompilationAssertions.kt:111)
	at arrow.meta.plugin.testing.CompilationAssertionsKt$interpreter$1.invoke(CompilationAssertions.kt)
	at arrow.meta.plugin.testing.CompilerTest.run$meta_test(CompilerTestDSL.kt:55)
	at arrow.meta.plugin.testing.CompilationAssertionsKt.assertThis(CompilationAssertions.kt:61)
	at example.HelloWorldPluginTest.test(HelloWorldPluginTest.kt:44)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
```

**To Reproduce**
Using the following code, the test "test" will fail

```
class HelloWorldPluginTest {

    companion object {

        val initialCode = """|fun log() {
              |     log("TEST", "this is a test")
              |}
              |
              |fun log(tag: String, message: String) { }
            """.trimMargin().trim()

        val expectedOutput = """|fun log() {
              |     log("TEST", "this is a test changed by Arrow Meta")
              |}
              |
              |fun log(tag: String, message: String) { }
            """.trimMargin().trim()
    }

    @Test
    fun `checks that compiles`() {
        assertThis(CompilerTest(
            code = {
                """
        | fun hello(): String =
        |   "Hello world!"
        | 
        """.source
            },
            assert = {
                compiles
            }
        ))
    }

    @Test
    fun `test`() {
        assertThis(CompilerTest(
            config = { listOf(addMetaPlugins(MetaPlugin())) },
            code = { initialCode.source },
            assert = { quoteOutputMatches(expectedOutput.source) }
        ))
    }

}

```

I can see the next logs:

```
v: Output:
/var/folders/2c/v1qydxqd3h5fdlqzb671vkhr9lc6zs/T/Kotlin-Compilation3027256973569121546/classes/SourceKt.class
Sources:
/var/folders/2c/v1qydxqd3h5fdlqzb671vkhr9lc6zs/T/Kotlin-Compilation3027256973569121546/sources/Source.kt
v: Output:
/var/folders/2c/v1qydxqd3h5fdlqzb671vkhr9lc6zs/T/Kotlin-Compilation3027256973569121546/classes/META-INF/main.kotlin_module
Sources:
/var/folders/2c/v1qydxqd3h5fdlqzb671vkhr9lc6zs/T/Kotlin-Compilation3027256973569121546/sources/Source.kt

```
but at some point it will look for Source.kt.meta file causing the test to fail.

**Expected behavior**
The test does not fail with a FileNotFoundException.

**Environment (please complete the following information):**
 - Kotlin version: 1.4.10 
- Intellij IDEA version: Android Studio 4.0.1

**Comment by rachelcarmena** [comment]
Hi @ocampoleandro :wave: Thanks for your interest!! 

`quoteOutputMatches` is used for checking the quote output during the compilation. 

It's necessary to use:

```
//metadebug
```

to collect that output into `.kt.meta` file.

It's useful to check that everything is working as expected during the compilation and you can find other examples in `compiler-plugin` tests.

**Comment by ocampoleandro** [comment]
Hi @rachelcarmena .

Thanks for the explanation.

I have one more question related to testing. How can I run tests with some classes already created?. 
I have tried with classes that are part of my classpath and when I run the test it shows "unresolved reference".

For example: 
```
import example.MyClass
```

will throw unresolved reference: example.

Also I noticed the same with dependencies.  In this case I tried to use
```
CompilerTest.addDependencies(Dependency("someArtifactId:version"))
```
and it throws the same error.

Also I checked the [compile testing](https://github.com/tschuchortdev/kotlin-compile-testing) library used for this and it says:
_Inherit classpath: Compiled sources have access to classes in your application_

So I guess I missing something?

I checked the examples in arrow-meta:meta-test but I could not find any example related to this.
